### PR TITLE
Clean up EmailField

### PR DESCRIPTION
**Pull request by assertchris**

### CODE REVIEW

File: forms/EmailField.php [hunk 1]
```diff
@@ -1,54 +1,68 @@
...
- * 
+ * Text input field with validation for correct email format according to RFC 2822.
+ *
  * @package forms
  * @subpackage fields-formattedinput
  */
 class EmailField extends TextField {
-
+	/**
+	 * @return string
```
Comment (tractorcow) [inline forms/EmailField.php]: This phpdoc is already on FormField::Type(), but with more documentation. You probably don't need to duplicate the whole block all the way down the inheritance chain; PHPDoc is by nature already inherited.

File: forms/EmailField.php [hunk 2]
```diff
@@ -1,54 +1,68 @@
...
+	 */
 	public function Type() {
 		return 'email text';
 	}
 
+	/**
+	 * @return array
```
Comment (tractorcow) [inline forms/EmailField.php]: Already documented on FormField::getAttributes, as above.

File: forms/EmailField.php [hunk 3]
```diff
@@ -1,54 +1,68 @@
...
 
-		$pcrePattern = '^[a-z0-9!#$%&\'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&\'*+/=?^_`{|}~-]+)*'
-			. '@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$';
+		$head = '[a-z0-9!#$%&\'*+/=?^_`{|}~-]+';
+		$tail = '[a-z0-9](?:[a-z0-9-]*[a-z0-9])?';
 
-		// PHP uses forward slash (/) to delimit start/end of pattern, so it must be escaped
-		$pregSafePattern = str_replace('/', '\\/', $pcrePattern);
+		$pattern = sprintf(
+			'^%s(?:\\.%s)*@(?:%s\\.)+%s$',
```
Comment (tractorcow) [inline forms/EmailField.php]: I'm not sure that this increases the readability of the regexp.

**Comment by assertchris** [comment]
@tractorcow poke!

File: forms/EmailField.php [hunk 4]
```diff
@@ -1,54 +1,59 @@
...
- * 
+ * Text input field with validation for correct email format according to RFC 2822.
+ *
  * @package forms
  * @subpackage fields-formattedinput
  */
 class EmailField extends TextField {
-
+	/**
+	 * {@inheritdoc}
```
Comment (tractorcow) [inline forms/EmailField.php]: Redundant?

Comment (assertchris) [inline forms/EmailField.php]: I don't agree. Knowing there's PHPDoc info on a parent is better than not knowing. It also creates a habit of strictly opt-out PHPDocs.

**Comment by assertchris** [comment]
@tractorcow ping

**Pull request merged**